pkg/ruler/appender.go

```diff
@@ -0,0 +1,179 @@
+package ruler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/rules"
+	"github.com/prometheus/prometheus/storage"
+
+	"github.com/grafana/loki/pkg/util"
+)
+
+type RemoteWriteAppendable struct {
+	groupAppender map[string]*RemoteWriteAppender
+
+	userID    string
+	cfg       Config
+	overrides RulesLimits
+	logger    log.Logger
+}
+
+func newRemoteWriteAppendable(cfg Config, overrides RulesLimits, logger log.Logger, userID string) *RemoteWriteAppendable {
+	return &RemoteWriteAppendable{
+		logger:        logger,
+		userID:        userID,
+		cfg:           cfg,
+		overrides:     overrides,
+		groupAppender: make(map[string]*RemoteWriteAppender),
+	}
+}
+
+type RemoteWriteAppender struct {
+	logger       log.Logger
+	ctx          context.Context
+	remoteWriter remoteWriter
+	userID       string
+	groupKey     string
+
+	queue *util.EvictingQueue
+}
+
+func (a *RemoteWriteAppendable) Appender(ctx context.Context) storage.Appender {
+	groupKey := retrieveGroupKeyFromContext(ctx)
+
+	// create or retrieve an appender associated with this groupKey (unique ID for rule group)
+	appender, found := a.groupAppender[groupKey]
+	if found {
+		return appender
```

Comment:
Since this is ultimately called on every evaluation cycle in the Ruler, we can probably add a method `WithCapacity` which can mutate an appenders capacity and thus will be kept up to date with the overrides (would need a test)
```suggestion
		return appender.WithCapacity(capacity)
```